ecs-cli/modules/utils/compose/convert_task_definition_test.go

```diff
@@ -731,6 +731,160 @@ task_definition:
+		CPU:               int64(2),
+		Memory:            int64(3),
+		MemoryReservation: int64(3),
+	}
+
+	// define ecs-params values we expect to override containerConfig vals
+	webCPU := int64(5)
+	webMem := int64(15)
+	webMemRes := int64(10)
+
+	ecsParamsString := `version: 1
+task_definition:
+  services:
+    web:
+      cpu_shares: 5
+      mem_limit: 15m
+      mem_reservation: 10m`
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs fields tempfile")
+
+	defer os.Remove(tmpfile.Name())
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs fields tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+	ecsParams, err := ReadECSParams(ecsParamsFileName)
+	assert.NoError(t, err, "Could not read ECS Params file")
+
+	containerConfigs := []adapter.ContainerConfig{*containerConfig}
+	taskDefinition, err := convertToTaskDefWithEcsParamsInTest(t, containerConfigs, "", ecsParams)
+
+	containerDefs := taskDefinition.ContainerDefinitions
+	web := findContainerByName("web", containerDefs)
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, webCPU, aws.Int64Value(web.Cpu), "Expected CPU to match")
+		assert.Equal(t, webMem, aws.Int64Value(web.Memory), "Expected Memory to match")
+		assert.Equal(t, webMemRes, aws.Int64Value(web.MemoryReservation), "Expected MemoryReservation to match")
+	}
+}
+
+func TestConvertToTaskDefinitionWithECSParams_NoMemoryProvided(t *testing.T) {
+	containerConfig := &adapter.ContainerConfig{
+		Name:  "web",
+		Image: "httpd",
+	}
+
+	// define ecs-params values we expect to override containerConfig vals
+	webCPU := int64(5)
+
+	ecsParamsString := `version: 1
+task_definition:
+  services:
+    web:
+      cpu_shares: 5`
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs fields tempfile")
+
+	defer os.Remove(tmpfile.Name())
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs fields tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+	ecsParams, err := ReadECSParams(ecsParamsFileName)
+	assert.NoError(t, err, "Could not read ECS Params file")
+
+	containerConfigs := []adapter.ContainerConfig{*containerConfig}
+	taskDefinition, err := convertToTaskDefWithEcsParamsInTest(t, containerConfigs, "", ecsParams)
+
+	containerDefs := taskDefinition.ContainerDefinitions
+	web := findContainerByName("web", containerDefs)
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, webCPU, aws.Int64Value(web.Cpu), "Expected CPU to match")
+		assert.Equal(t, int64(defaultMemLimit), aws.Int64Value(web.Memory), "Expected Memory to match default")
+		assert.Empty(t, aws.Int64Value(web.MemoryReservation), "Expected MemoryReservation to be empty")
+	}
+}
```

Comment:
Can you maybe add a test in which services are specified in the ecs-params file but not all the fields are specified so that it still sets the values from the ContainerConfig?

Reply:
done